crypto/src/note.rs

```diff
@@ -90,6 +102,111 @@ impl Note {
         self.value
     }
 
+    /// Encrypt a note, returning its ciphertext.
+    pub fn encrypt(&self, esk: &ka::Secret) -> Result<[u8; NOTE_CIPHERTEXT_BYTES], Error> {
+        let epk = esk.diversified_public(&self.diversified_generator());
+        let shared_secret = esk
+            .key_agreement_with(&self.transmission_key())
+            .map_err(|_| Error::EncryptionError)?;
+
+        // Use Blake2b-256 to derive encryption key.
+        let mut kdf_params = blake2b_simd::Params::new();
+        kdf_params.hash_length(32);
+        let mut kdf = kdf_params.to_state();
+        kdf.update(&shared_secret.0);
+        kdf.update(&epk.0);
+        let kdf_output = kdf.finalize();
+        let key = Key::from_slice(kdf_output.as_bytes());
```

Comment:
if this is common with the note encryption, could it be extracted as a standalone function?